src/diffusers/textual_inversion_utils.py

```diff
@@ -0,0 +1,101 @@
+        the tokenizer's vocabulary and the text encoder's embedding matrix.
+
+        Iters through the dictionary and adds the token to the tokenizer's vocabulary and the embedding to the text
+        encoder's embedding matrix.
+        """
+
+        if isinstance(embeddings, dict):
+            for token, embedding_path in embeddings.items():
+                # check if token in tokenizer vocab
+                # if yes, raise exception
+                if token in self.tokenizer.get_vocab():
+                    raise ValueError(
+                        f"Token {token} already in tokenizer vocabulary. Please choose a different token name."
+                    )
+
+                embedding_dict = torch.load(embedding_path)
+                embedding = self.extract_embedding_from_dict(embedding_dict)
+
+                self.add_textual_inversion_embedding(token, embedding)
+
+        elif isinstance(embeddings, list):
+            for embedding_path in embeddings:
+                embedding_dict = torch.load(embedding_path)
+                token = self.extract_token_from_dict(embedding_dict)
+                embedding = self.extract_embedding_from_dict(embedding_dict)
+
+                # check if token in tokenizer vocab
+                # if yes, raise exception
+                if token in self.tokenizer.get_vocab():
+                    raise ValueError(
+                        f"Token {token} already in tokenizer vocabulary. Please choose a different token name."
+                    )
+                self.add_textual_inversion_embedding(token, embedding)
+
+    def extract_embedding_from_dict(self, embedding_dict):
+        r"""
+        Extracts the embedding from the embedding dictionary.
+        """
+        # auto1111 embedding case
+        if "string_to_param" in embedding_dict:
+            embedding_dict = embedding_dict["string_to_param"]
+            embedding = embedding_dict["*"]
+            return embedding
+
+        return list(embedding_dict.values())[0]
+
+    def extract_token_from_dict(self, embedding_dict):
+        r"""
+        Extracts the token from the embedding dictionary.
+        """
+        # auto1111 embedding case
+        if "string_to_param" in embedding_dict:
+            token = embedding_dict["name"]
+            return token
+
+        return list(embedding_dict.keys())[0]
+
+    def add_textual_inversion_embedding(self, token, embedding):
+        r"""
+        Adds a token to the tokenizer's vocabulary and an embedding to the text encoder's embedding matrix.
+        """
+        # check if token in tokenizer vocab
+        # if yes, raise exception
+        if token in self.tokenizer.get_vocab():
+            raise ValueError(f"Token {token} already in tokenizer vocabulary. Please choose a different token name.")
+
+        embedding = embedding.to(self.text_encoder.device)
```

Comment:
I think you can get rid of this by using map_location in torch.load.

Reply:
I implemented this.